upgrade/boot_loader/VerifierEcDsa.cpp

```diff
@@ -1,29 +1,67 @@
 #include "upgrade/boot_loader/VerifierEcDsa.hpp"
+#include "infra/util/ByteRange.hpp"
+#include "infra/util/MemoryRange.hpp"
 #include "mbedtls/sha256.h"
 #include "uECC.h"
 
 namespace application
 {
     VerifierEcDsa::VerifierEcDsa(infra::ConstByteRange key)
         : key(key)
-    {
-        assert(key.size() == 56);
-    }
+    {}
 
     bool VerifierEcDsa::IsValid(hal::SynchronousFlash& flash, const hal::SynchronousFlash::Range& signature, const hal::SynchronousFlash::Range& data) const
     {
         std::array<uint8_t, 32> messageHash = Hash(flash, data);
-        std::array<uint8_t, 56> storedSignature;
+        auto storedSignature = GetSignatureStorage(flash, signature);
 
-        if (signature.second - signature.first != storedSignature.size())
+        if (storedSignature.size() == 0)
             return false;
 
-        flash.ReadBuffer(storedSignature, signature.first);
-
         return uECC_verify(key.begin(),
                    messageHash.data(),
                    messageHash.size(),
-                   storedSignature.data(),
-                   uECC_secp224r1()) == 1;
+                   storedSignature.begin(),
+                   GetCurve()) == 1;
+    }
+
+    VerifierEcDsa224::VerifierEcDsa224(infra::ConstByteRange key)
+        : VerifierEcDsa(key)
+    {
+        assert(key.size() == signSize);
+    }
+
+    infra::ConstByteRange VerifierEcDsa224::GetSignatureStorage(hal::SynchronousFlash& flash, const hal::SynchronousFlash::Range& signature) const
+    {
+        if (signature.second - signature.first != signatureStorage.size())
+            return infra::ConstByteRange();
+
+        flash.ReadBuffer(infra::ConstCastByteRange(signatureStorage), signature.first);
```

Comment:
The function is named "GetSignatureStorage", but next to returning the storage, it also reads it from flash. I would either change the name of the function, or move reading from flash outside of this function